src/turbomind/models/llama/LlamaBatch.cc

```diff
@@ -1199,6 +1202,15 @@ auto LlamaBatch<T>::Finish(GenerationState& g) -> std::vector<Signal>
         }
     }
 
+    // Cache computed blocks to block trie
+    for (int i = 0; i < batch_size; ++i) {
+        auto& seq = *state_->sequences[i];
+        // only cache prompt blocks
+        if (seq.input_length > 1) {
```

Comment:
Using `seq.input_length > 1` to determine if it is a prompt has certain limitations. In the future, when supporting speculative decoding, input_length greater than 1 may also be in the decoding stage.
A feasible way, for example, is to add a variable in the `Sequence` to indicate whether it is a prompt and if it is the prompt in the first round of multi-turn dialogue.

Reply:
I added the `stage` in sequence to determine `prefill` or `decoding` and added `interact_count` to record the interact count in multi-turns dialogue (stateful inference).